STYLE-GUIDE.md

```diff
@@ -0,0 +1,237 @@
+# Spectrometer style guide
+
+[Back to development doc homepage](devdocs/index.md)
+
+Through a number of pull requests, reviews, and discussions, a codebase style has emerged.  This document is an attempt to capture that, and prevent further discussions that slow down work unnecessarily.
+
+This style guide is forked from Github's [semantic](https://github.com/github/semantic).
+
+## General guidelines
+
+Make your code look like the code around it. Consistency is the name of the game.
+
+Use `fourmolu` for formatting.  Our CI setup enforces that all changes pass a `fourmolu` run with no differences.
+
+Our CI systems ensure that all patches pass `hlint`'s muster. We have our own set of rules in `.hlint.yaml`.
+
+We strongly recommend adding Haddock documentation to any top-level declaration, unless its purpose is immediately apparent from its name.
+Comments should describe the "why", type signatures should describe the "what", and the code should describe the "how".
+
+## Formatting
+
+2 spaces everywhere. Tabs are forbidden.  These are enforced by the formatter.
+
+Haskell indentation can be unpredictable, so generally stick with what your editor suggests.
+There is no hard line-length limit, though if you go beyond 110 or 120 you should generally split it up, especially for type signatures.
+
+### Use applicative notation when constructing simple data types
+
+``` haskell
+thing :: Parser Foo
+
+-- Bad:
+thing = do
+  a <- bar
+  b <- baz
+  pure (Foo a b)
+
+-- Better:
+thing = Foo <$> bar <*> baz
+```
+
+Code written with applicative notation is less powerful than code written in `do`-notation (since that code is, by definition, monadic).  As a result, it can be easier to reason about, as long as it's also readable.
+
+Over-reliance on applicative notation can create code that is difficult to read. Don't use applicative notation in combination with operator sections. If in doubt, write it with `do` notation and see if it's more immediately comprehensible.
```

Comment:
>Don't use applicative notation in combination with operator sections

Can you include an example for this?

Reply:
No, this should be reworded so that an example isn't necessary.